packages/@react-native-windows/telemetry/src/telemetry.ts

```diff
@@ -4,207 +4,411 @@
  * @format
  */
 
-import path from 'path';
-import {randomBytes} from 'crypto';
 import * as appInsights from 'applicationinsights';
-import {execSync} from 'child_process';
-import {CodedError} from './CodedError';
 
+import * as basePropUtils from './utils/basePropUtils';
+import * as versionUtils from './utils/versionUtils';
+import * as errorUtils from './utils/errorUtils';
+import * as projectUtils from './utils/projectUtils';
+
+export interface TelemetryOptions {
+  setupString: string;
+  preserveErrorMessages: boolean;
+}
+
+export interface CommandStartInfo {
+  commandName: string;
+  args: Record<string, any>;
+  options: Record<string, any>;
+  defaultOptions: Record<string, any>;
+}
+
+export interface CommandEndInfo {
+  resultCode: errorUtils.CodedErrorType;
+}
+
+interface CommandInfo {
+  startTime?: number;
+  endTime?: number;
+  startInfo?: CommandStartInfo;
+  endInfo?: CommandEndInfo;
+}
+
+const DefaultSetupString = '795006ca-cf54-40ee-8bc6-03deb91401c3';
+
+export const NpmPackagesWeTrack: string[] = [
+  '@react-native-community/cli',
+  '@react-native-windows/cli',
+  '@react-native-windows/telemetry',
+  'react',
+  'react-native',
+  'react-native-windows',
+  'react-native-windows-init',
+];
+
+export const NuGetPackagesWeTrack: string[] = [
+  'Microsoft.UI.Xaml',
+  'Microsoft.Windows.CppWinRT',
+  'Microsoft.WinUI',
+];
+
+/**
+ * The Telemetry class is responsible for reporting telemtery for RNW CLI.
+ */
 export class Telemetry {
-  static client?: appInsights.TelemetryClient | undefined = undefined;
+  protected static client?: appInsights.TelemetryClient = undefined;
+  protected static options: TelemetryOptions = {
+    setupString: DefaultSetupString,
+    preserveErrorMessages: false,
+  };
 
-  static disable() {
+  protected static isTest: boolean = basePropUtils.isCliTest();
+  protected static commandInfo: CommandInfo = {};
+  protected static versionsProp: Record<string, string> = {};
+  protected static projectProp?:
+    | projectUtils.AppProjectInfo
+    | projectUtils.DependencyProjectInfo = undefined;
+
+  protected static reset(): void {
+    // Reset client
     if (Telemetry.client) {
-      Telemetry.client.config.disableAppInsights = true;
+      Telemetry.client.flush({isAppCrashing: true});
```

Comment:
This call (with `isAppCrashing`) is only made in-between tests to make sure events don't leak across.

The telemetry library was designed for long-running web apps, so batches up telemetry events and schedules them for later. So if your app is coming down because of a crash, this flag lets you tell the telemetry to *really* start flushing right away. Technically you can also pass in a callback that gets called when the telemetry has been submitted, but I omitted it here, since we're just resetting the state for the next test.

Reply:
I removed this.